git-config/src/file/git_config.rs

```diff
@@ -272,14 +277,37 @@ impl<'event> GitConfig<'event> {
         config_path: Option<&std::path::Path>,
         options: &from_paths::Options,
     ) -> Result<(), from_paths::Error> {
-        // TODO
-        fn include_condition_is_true(_condition: &Cow<str>, _options: &from_paths::Options) -> bool {
+        // TODO handle new git undocumented feature `hasconfig`
+        fn include_condition_is_true(
+            condition: &Cow<str>,
+            target_config_path: &Option<&Path>,
+            options: &from_paths::Options,
+        ) -> bool {
+            if let Some(git_dir) = options.git_dir {
+                if let Some(condition) = condition.strip_prefix("gitdir:") {
+                    let path = values::Path::from(Cow::Borrowed(condition.as_bytes()));
+                    if let Ok(resolved) = resolve(path, target_config_path, options) {
+                        return resolved == git_dir;
+                    }
+                    return false;
+                } else if let Some(_condition) = condition.strip_prefix("gitdir/i:") {
+                    todo!()
+                }
+            }
+            if let Some(branch_name) = options.branch_name {
+                if branch_name.category() == Some(git_ref::Category::LocalBranch) {
+                    let branch_name = branch_name.shorten();
+                    if let Some(condition) = condition.strip_prefix("onbranch:") {
+                        return condition == branch_name;
+                    }
+                }
+            }
             false
         }
 
         fn resolve_includes_recursive(
             target_config: &mut GitConfig,
-            target_config_path: Option<&Path>,
+            target_config_path: &Option<&Path>,
```

Comment:
Options should never be passed as reference, but should always be owned. That way they are owned where they are used, which doesn't affect their content at all.

Reply:
> Options should never be passed as reference, but should always be owned

Is this a general rule / Do you mean that since its Option of reference there is a cheap copy of it so no point to pass by reference similar to not passing reference to e.g. i32?

Reply:
Yes, everything that that is `Copy` should be copied in. For me it's only a rule because the callee can do more if its values are owned. This is particularly true for `Option` which is `Copy` if its contents is copy. And `Some(&_)` are always copy as shared borrows are copy.